app/code/local/Magmodules/Webwinkelconnect/Helper/Data.php

```diff
@@ -143,6 +143,50 @@ public function syncProductReview(array $product_review): array {
         }
     }
 
+
+    public function hasConsent(Mage_Sales_Model_Order $order): bool {
+        $permission_url = 'https://dashboard.webwinkelkeur.nl/api/2.0/order_permissions.json?' .
+            http_build_query([
+                'orderNumber' => $order->getIncrementId(),
+                'id' => $this->getShopId(),
+                'code' => $this->getApiKey(),
+            ]);
+
+        $start_time = microtime(true);
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_TIMEOUT, 10);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+        curl_setopt($ch, CURLOPT_FAILONERROR, true);
+        curl_setopt($ch, CURLOPT_URL, $permission_url);
+        try {
+            $response = curl_exec($ch);
+            $message = '';
+            if ($response === false) {
+                $message = curl_error($ch);
```

Comment:
You should expand the message a bit: what exactly went wrong, and include the error code.

```suggestion
                $message = sprintf("Consent check failed with cURL error: (%s) %s", curl_errno($ch), curl_error($ch));
```